features/tailormade_profile/src/main/java/com/future/tailormade_profile/feature/editProfile/view/EditProfileFragment.kt

```diff
@@ -0,0 +1,147 @@
+import com.future.tailormade.util.extension.toDateString
+import com.future.tailormade_profile.R
+import com.future.tailormade_profile.databinding.FragmentEditProfileBinding
+import com.future.tailormade_profile.feature.editProfile.viewModel.EditProfileViewModel
+import com.google.android.material.datepicker.MaterialDatePicker
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.InternalCoroutinesApi
+
+@AndroidEntryPoint
+class EditProfileFragment : BaseFragment() {
+
+  private val viewModel: EditProfileViewModel by viewModels()
+
+  private var birthDate: Long = 0L
+
+  private lateinit var binding: FragmentEditProfileBinding
+
+  private lateinit var birthDatePicker: MaterialDatePicker<Long>
+
+  override fun getLogName(): String =
+      "com.future.tailormade_profile.feature.editProfile.view.EditProfileFragment"
+
+  override fun getScreenName(): String = "Edit Profile"
+
+  @ExperimentalCoroutinesApi
+  @InternalCoroutinesApi
+  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+      savedInstanceState: Bundle?): View? {
+    setupDatePicker()
+    binding = FragmentEditProfileBinding.inflate(inflater, container, false)
+
+    with(binding) {
+      buttonSubmitEditProfileForm.setOnClickListener {
+        submitForm(editTextNameEditProfile.text.toString(),
+            editTextPhoneNumberEditProfile.text.toString(),
+            editTextBirthDateEditProfile.text.toString(),
+            editTextLocationEditProfile.text.toString())
+      }
+
+      editTextLocationEditProfile.debounceOnTextChanged(
+          viewModel.viewModelScope, viewModel::updateLocations)
+
+      textInputBirthDateEditProfile.setEndIconOnClickListener {
+        showDatePicker()
+      }
+    }
+
+    return binding.root
+  }
+
+  override fun setupFragmentObserver() {
+    super.setupFragmentObserver()
+    setupProfileDataObserver()
+    setupLocationObserver()
+  }
+
+  private fun isFormValid(name: String, birthDate: String, phoneNumber: String?) =
+      name.isNotBlank() && birthDate.isNotBlank() && (phoneNumber?.isPhoneNumberValid() ?: true)
+
+  private fun setFormErrorMessage() {
+    with(binding) {
+      textInputNameEditProfile.error = when {
+        editTextNameEditProfile.text.toString().isBlank() -> Constants.NAME_IS_EMPTY
+        else -> null
+      }
+
+      textInputPhoneNumberEditProfile.error = when {
+        editTextPhoneNumberEditProfile.text.toString().isPhoneNumberValid().not() -> Constants.PHONE_NUMBER_IS_NOT_VALID
+        else -> null
+      }
+
+      textInputBirthDateEditProfile.error = when {
+        editTextBirthDateEditProfile.text.toString().isBlank() -> Constants.BIRTH_DATE_IS_NOT_SET
+        else -> null
+      }
+    }
+  }
+
+  private fun setupDatePicker() {
+    birthDatePicker = MaterialDatePicker.Builder.datePicker().setTitleText(
+        Constants.BIRTH_DATE_PICKER_TITLE).build()
```

Comment:
Why constant? how about put this on string?

Reply:
It used several times kak, oh use strings.xml kak?